src/main/java/Controller/playAudioController.java

```diff
@@ -24,9 +24,9 @@ public playAudioController(playAudioInputBoundary playAudioInteractor) {
      * Function calling the use case for playing audio
      * @param songFile is the File object of the MP3 file
      */
-    public void playAudio(File songFile) {
-        audioInputData audioInput = new audioInputData(songFile);
-        this.playAudioInteractor.playAudio(audioInput);
+    public void playAudio(songInputData songFile) {
+        songInputData songInput = new songInputData(songFile.getSong());
```

Comment:
Why are you creating a new songInputData when the parameter passed to playAudio is a songInputData already?

Reply:
Also, what's the difference between playAudioController and Raf's musicEngineController?